### PR TITLE
Add errorlint and fix all existing issues

### PR DESCRIPTION
**Pull request by markusthoemmes**
As per title, this seems quite useful to check imo.

/hold
Until linting version bump comes through.

### CODE REVIEW

**Review by vagababov — COMMENTED** [review]
/lgtm

**Review by vagababov — COMMENTED** [review]
/lgtm

File: testutils/clustermanager/e2e-tests/common/common_test.go [hunk 1]
```diff
@@ -68,7 +67,7 @@ func TestGetRepoName(t *testing.T) {
 			"a/b/c/", nil, "c", nil,
 		}, {
 			// Git error
-			"", fmt.Errorf("git error"), "", fmt.Errorf("failed git rev-parse --show-toplevel: 'git error'"),
+			"", fmt.Errorf("git error"), "", fmt.Errorf("failed git rev-parse --show-toplevel: git error"),
```
Comment (julz) [inline testutils/clustermanager/e2e-tests/common/common_test.go:70]: kind of surprised these didn't fail a linter for using Errorf instead of errors.New 🤔

**Review by julz — APPROVED** [review]
/lgtm

**Pull request merged**